tests/lookup/tests.py

```diff
@@ -1357,6 +1357,9 @@ def setUpTestData(cls):
         cls.s1 = Season.objects.create(year=1942, gt=1942)
         cls.s2 = Season.objects.create(year=1842, gt=1942, nulled_text_field="text")
         cls.s3 = Season.objects.create(year=2042, gt=1942)
+        cls.g1 = Game.objects.create(season=cls.s1, home="NY", away="Boston")
+        cls.g2 = Game.objects.create(season=cls.s1, home="NY", away="Tampa")
+        cls.g3 = Game.objects.create(season=cls.s3, home="Boston", away="Tampa")
```

Comment:
To guard against the “home=NY & away=Boston” filter incorrectly turning into an OR as happened in ticket 34450, we need another entry that OR would select but AND would not
```suggestion
        cls.g3 = Game.objects.create(season=cls.s3, home="Boston", away="Tampa")
        cls.g4 = Game.objects.create(season=cls.s3, home="Tampa", away="Boston")
```

Reply:
@rdaysky the tests below already fail without the patch applied in their current form so I'll let the mergers decide if they want to add extra coverage. I don't see harm in doing so but we usually only ship the minimal cases to reproduce.